content/terraform/best-practices.md

```diff
@@ -298,9 +304,20 @@ generate meaningful documentation.
 
 # Module Usage
 
-## Use fully-qualified git URLs
+## Use Terraform registry format with exact version numbers
+
+There are many ways to express a module's source. Our convention is to use Terraform registry syntax with an explicit version.
+
+```
+  source  = "cloudposse/label/null"
+  version = "0.22.0" 
+```
+
+The reason to pin to an explicit version rather than a range like `>= 0.22.0` is that any update is capable of breaking something. Any changes to your infrastructure should be implemented and reviewed under your control, not blindly automatic based on when you deployed it.
+
+{{% dialog type="info" icon="fa-info-circle" title="Note" %}}
 
-There are many ways to express a module's source. Our convention is to use the pure git url. 
+Prior to Terraform v0.13, our convention was to use the pure git url. 
```

Comment:
```suggestion
Prior to Terraform v0.13, our convention was to use the pure git URL (e.g. `git::https://github.com/...?ref=tags/...`). We no longer use this format and our automated tests will deliberately error if the format is detected.
```

Reply:
At the moment we do not reject that. We only check to make sure you have a version and not `master`